src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/helpers/terraform/adf_terraform.sh

```diff
@@ -0,0 +1,175 @@
+#!/usr/bin/env bash
+PATH=$PATH:$(pwd)
+export PATH
+CURRENT=$(pwd)
+terraform --version 
+echo "Terraform stage: $TF_STAGE"
+
+tfinit(){
+    # retrieve regional S3 bucket name from parameter store
+    S3_BUCKET_REGION_NAME=$(aws ssm get-parameter --name /cross_region/s3_regional_bucket/"$AWS_REGION" --region "$AWS_DEFAULT_REGION" | jq .Parameter.Value | sed s/\"//g)
+    mkdir -p "$CURRENT"/tmp/"$TF_VAR_TARGET_ACCOUNT_ID"-"$AWS_REGION"
+    cd "$CURRENT"/tmp/"$TF_VAR_TARGET_ACCOUNT_ID"-"$AWS_REGION" || exit
+    cp -R "$CURRENT"/tf/* "$CURRENT"/tmp/"$TF_VAR_TARGET_ACCOUNT_ID"-"$AWS_REGION"
+    # if account related variables exist copy the folder in the work directory
+    if [ -d "$CURRENT/tfvars/$TF_VAR_TARGET_ACCOUNT_ID" ]; then
+        cp -R "$CURRENT"/tfvars/"$TF_VAR_TARGET_ACCOUNT_ID"/* "$CURRENT"/tmp/"$TF_VAR_TARGET_ACCOUNT_ID"-"$AWS_REGION"
+    fi
+    cp -R "$CURRENT"/tfvars/global.auto.tfvars "$CURRENT"/tmp/"$TF_VAR_TARGET_ACCOUNT_ID"-"$AWS_REGION"
+    terraform init \
+        -backend-config "bucket=$S3_BUCKET_REGION_NAME" \
+        -backend-config "region=$AWS_REGION" \
+        -backend-config "key=$ADF_PROJECT_NAME/$ACCOUNT_ID.tfstate" \
+        -backend-config "dynamodb_table=adf-tflocktable"
+
+    echo "Bucket: $S3_BUCKET_REGION_NAME"
+    echo "Region: $AWS_REGION"
+    echo "Key:    $ADF_PROJECT_NAME/$ACCOUNT_ID.tfstate"
+    echo "DynamoDB table: adf-tflocktable"
+}
+tfplan(){  
+    bash "$CURRENT"/adf-build/helpers/sts.sh "$TF_VAR_TARGET_ACCOUNT_ID" "$TF_VAR_TARGET_ACCOUNT_ROLE"
+    terraform plan -out "$ADF_PROJECT_NAME"-"$TF_VAR_TARGET_ACCOUNT_ID"
+}
+tfapply(){
+    terraform apply "$ADF_PROJECT_NAME"-"$TF_VAR_TARGET_ACCOUNT_ID"
+}
+
+# if REGIONS is not defined as pipeline parameters use default region
```

Comment:
If the regions are not defined it should use the deployment region, which is the region where this script runs.

Reply:
In the script I put to define the default region

if [[ -z "$REGIONS" ]]
then
    REGIONS=$AWS_DEFAULT_REGION
fi